File: library/src/main/java/cc/shinichi/library/glide/cache/DataCacheKey.java
package cc.shinichi.library.glide.cache;

import com.bumptech.glide.load.Key;

import java.security.MessageDigest;

/**
 * @author 工藤
 * @email 18883840501@163.com
 * com.fan16.cn.loader.glide
 * create at 2018/5/10  11:12
 * description:
 */
public class DataCacheKey implements Key {

    private final Key sourceKey;
    private final Key signature;

    public DataCacheKey(Key sourceKey, Key signature) {
        this.sourceKey = sourceKey;
        this.signature = signature;
    }

    public Key getSourceKey() {
        return sourceKey;
    }

    @Override
    public boolean equals(Object o) {
        if (o instanceof DataCacheKey) {
            DataCacheKey other = (DataCacheKey) o;
            return sourceKey.equals(other.sourceKey) && signature.equals(other.signature);
        }
        return false;
    }

    @Override
    public int hashCode() {
        int result = sourceKey.hashCode();
        result = 31 * result + signature.hashCode();
        return result;
    }

    @Override
    public String toString() {
        return "DataCacheKey{" + "sourceKey=" + sourceKey + ", signature=" + signature + '}';
    }

    @Override
    public void updateDiskCacheKey(MessageDigest messageDigest) {
        sourceKey.updateDiskCacheKey(messageDigest);
        signature.updateDiskCacheKey(messageDigest);
    }
}

File: library/src/main/java/cc/shinichi/library/glide/cache/DataCacheKey.kt
package cc.shinichi.library.glide.cache

import com.bumptech.glide.load.Key
import java.security.MessageDigest

/**
 * @author 工藤
 * @email qinglingou@gmail.com
 * create at 2018/5/10  11:12
 * description:
 */
class DataCacheKey(val sourceKey: Key, private val signature: Key) : Key {

    override fun equals(o: Any?): Boolean {
        if (o is DataCacheKey) {
            return sourceKey == o.sourceKey && signature == o.signature
        }
        return false
    }

    override fun hashCode(): Int {
        var result = sourceKey.hashCode()
        result = 31 * result + signature.hashCode()
        return result
    }

    override fun toString(): String {
        return "DataCacheKey{sourceKey=$sourceKey, signature=$signature}"
    }

    override fun updateDiskCacheKey(messageDigest: MessageDigest) {
        sourceKey.updateDiskCacheKey(messageDigest)
        signature.updateDiskCacheKey(messageDigest)
    }
}